test/unit/client.test.js

```diff
@@ -1019,6 +1058,171 @@ describe('message deletion', () => {
 	});
 });
 
+describe('delete for me functionality', () => {
+	const messageId = 'msg-delete-for-me-123';
+	let deletingUser, otherUser, deletingUserClient, otherUserClient;
+	let deletingUserEventSpy, otherUserEventSpy;
+
+	beforeEach(async () => {
+		deletingUser = { id: 'deleting-user', name: 'Deleting User' };
+		otherUser = { id: 'other-user', name: 'Other User' };
+
+		// Create clients for both users
+		deletingUserClient = await getClientWithUser(deletingUser);
+		otherUserClient = await getClientWithUser(otherUser);
+
+		// Set up event spies to track which events each client receives
+		deletingUserEventSpy = vi.fn();
+		otherUserEventSpy = vi.fn();
+
+		deletingUserClient.on('message.deleted', deletingUserEventSpy);
+		otherUserClient.on('message.deleted', otherUserEventSpy);
+
+		// Mock the delete API call to simulate server response
+		vi.spyOn(deletingUserClient, 'delete').mockResolvedValue({
+			message: {
+				id: messageId,
+				text: 'Test message',
+				user: deletingUser,
+				deleted_for_me: true,
+				deleted_at: new Date().toISOString(),
+			},
+		});
+	});
+
+	afterEach(() => {
+		vi.resetAllMocks();
+		deletingUserClient.off('message.deleted', deletingUserEventSpy);
+		otherUserClient.off('message.deleted', otherUserEventSpy);
+	});
+
+	it('should send delete_for_me=true to API when deleteForMe is true', async () => {
+		await deletingUserClient.deleteMessage(messageId, false, true);
+
+		expect(deletingUserClient.delete).toHaveBeenCalledWith(
+			`${deletingUserClient.baseURL}/messages/${encodeURIComponent(messageId)}`,
+			{ delete_for_me: true },
+		);
+	});
+
+	it('should handle message.deleted event with deleted_for_me=true for deleting user', () => {
```

Comment:
I believe this test is not testing what it declares. The test itself invokes WS event handlers for `deletingUserClient` but not for `otherUserClient`. It is normal that `otherUserEventSpy` will not be called if it is not registered with `deletingUserClient`. I think this test can be removed.